gammapy/scripts/catalog_query.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from ..utils.scripts import get_parser
+
+__all__ = ['catalog_query']
+
+def main(args=None):
+    parser = get_parser(catalog_query)
+    parser.add_argument('--catalog',
+                        help='Catalog for the source e.g. 3FGL')
+    parser.add_argument('--source',
+                        help='Source name e.g. J0349.9-2102')
+    parser.add_argument('--querytype',
```

Comment:
I'm not sure what the issue with `vars` is that you mentioned (I'm running into the different issue I pointed out for now).
But maybe you can try adding a choices option here (https://docs.python.org/3/library/argparse.html#choices) and if that doesn't work, try the debugging trick I mentioned of inserting `import IPython; IPython.embed()` here?

If you're still getting the error, please push the latest (rebased and other issues fixed) version and I'll have a look.